protocol/pubsub/v2/protocol.go

```diff
@@ -233,8 +233,14 @@ var _ protocol.Sender = (*Protocol)(nil)
 var _ protocol.Receiver = (*Protocol)(nil)
 var _ protocol.Closer = (*Protocol)(nil)
 
+type withInitialMessageAttributes struct{}
```

Comment:
More of a golang question... what's the advantage of making the context value's key a "struct" instead of just a "string" with a value of `initialMessageAttributes` ?

Reply:
> More of a golang question... what's the advantage of making the context value's key a "struct" instead of just a "string" with a value of `initialMessageAttributes` ?

See: https://pkg.go.dev/context#WithValue

> The provided key must be comparable and should not be of type string or any other built-in type to avoid collisions between packages using context. Users of WithValue should define their own types for keys. To avoid allocating when assigning to an interface{}, context keys often have concrete type struct{}. Alternatively, exported context key variables' static type should be a pointer or interface.

Reply:
Ah, thanks. That's interesting, sort of a "namespacing" thingy.